core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/iteration/IterateListSimultaneousltUnitTest.java

```diff
@@ -0,0 +1,74 @@
+            (numStr, nameStr) -> String.format("[Using Zip] Processing ArrayList Simultaneously : %d, %s", numStr, nameStr))
+            .collect(Collectors.toList());
+        Assertions.assertTrue(processedList.containsAll(
+          List.of("[Using Zip] Processing ArrayList Simultaneously : 1, John",
+                "[Using Zip] Processing ArrayList Simultaneously : 2, Nik",
+                "[Using Zip] Processing ArrayList Simultaneously : 3, Tom")));
+    }
+
+    @Test
+    public void givenTwoLists_whenIterateUsingIterator_thenProcessedSimultaneously() {
+        Iterator nameIterator = names.iterator();
+        Iterator numberIterator = numbers.iterator();
+        List<String> processedList = new ArrayList<>();
+        while (nameIterator.hasNext() && numberIterator.hasNext()) {
+            String processedData = String.format("[Using Iterator] Processing ArrayList Simultaneously : %d, %s", numberIterator.next(), nameIterator.next());
+            processedList.add(processedData);
+        }
+        Assertions.assertTrue(processedList.containsAll(
+          List.of("[Using Iterator] Processing ArrayList Simultaneously : 1, John",
+                "[Using Iterator] Processing ArrayList Simultaneously : 2, Nik",
+                "[Using Iterator] Processing ArrayList Simultaneously : 3, Tom")));
+    }
+
+    @Test
+    public void givenTwoLists_whenIterateUsingLoop_thenProcessedSimultaneously() {
+        List<String> processedList = new ArrayList<>();
+        for (int i = 0; i < names.size(); i++) {
+            String processedData = String.format("[Using Loop] Processing ArrayList Simultaneously : %d, %s", numbers.get(i), names.get(i));
+            processedList.add(processedData);
+        }
+        Assertions.assertTrue(processedList.containsAll(
+          List.of("[Using Loop] Processing ArrayList Simultaneously : 1, John",
+                "[Using Loop] Processing ArrayList Simultaneously : 2, Nik",
+                "[Using Loop] Processing ArrayList Simultaneously : 3, Tom")));
+    }
+
+    @Test
+    public void givenRecordList_whenIteratre_thenProcessedSimultaneously() {
+        List<EncapData> list = List.of(new EncapData("John", 1),
+                                new EncapData("Nik", 2),
+                                new EncapData("Tom", 3));
+        List<String> processedList = list.stream().map(l -> String.format("[Using Encapsulated Record] Processing ArrayList Simultaneously : %d, %s", l.number(), l.name()))
+          .collect(Collectors.toList());
+        Assertions.assertTrue(processedList.containsAll(
+          List.of("[Using Encapsulated Record] Processing ArrayList Simultaneously : 1, John",
+            "[Using Encapsulated Record] Processing ArrayList Simultaneously : 2, Nik",
+            "[Using Encapsulated Record] Processing ArrayList Simultaneously : 3, Tom")));
+    }
+
+}
+
+record EncapData (String name, int number){};
```

Comment:
Ah. It's Java `record` definition at the bottom of the unit test.

For portability, please define a `static class` within the test class and do it using a simple POJO, rather than a record, even though `record` is a better solution.